frontend/javascripts/dashboard/dataset/dataset_import_view.js

```diff
@@ -219,10 +233,20 @@ class DatasetImportView extends React.PureComponent<Props, State> {
       });
     }
 
-    let message;
+    let message = null;
     let type = "info";
-    const didParsingTheSavedDataSourceJSONSucceed = savedDataSourceOnServer.status == null;
-    if (!didParsingTheSavedDataSourceJSONSucceed) {
+    const applySuggestedSettings = () => {
+      this.props.form.setFieldsValue({
+        dataSourceJson: jsonStringify(inferredDataSource),
+      });
+      this.setState(currentState => {
+        const updatedStatus = _.cloneDeep(currentState.dataSourceSettingsStatus);
+        updatedStatus.appliedSuggestions = "YES";
```

Comment:
```suggestion
        const updatedStatus = {
          ...currentState.dataSourceSettingsStatus, appliedSuggestions: "YES"
        };
```

No need to get out the `_.cloneDeep` hammer I think :)